src/server/network.ts

```diff
@@ -61,12 +61,18 @@ export function rewriteCookies(cookies: types.SetNetworkCookieParam[]): types.Se
   });
 }
 
-function stripFragmentFromUrl(url: string): string {
+export function parsedURL(url: string): URL | null {
```

Comment:
Let's move all these methods to utils/ to share between client, server, cli and everyone.